docs/model-dev-guide/optimize-training.rst

```diff
@@ -0,0 +1,111 @@
+
+Please see :ref:`core-profiler` for details on enabling and configuring the Determined profiler for
+your experiment.
+
+The following system metrics are tracked:
+
+-  *GPU utilization (percent)*: current utilization of a GPU device
+-  *GPU free memory (bytes)*: current amount of free memory available on a GPU device
+-  *Network throughput - sent (bytes/s)*: change rate of number of bytes sent system-wide
+-  *Network throughput (received)*: change rate of number of bytes received system-wide
+-  *Disk IOPS (bytes/s)*: change rate of number of read + writes system-wide
+-  *Disk throughput - reads (bytes/s)*: change rate of bytes read system-wide
+-  *Disk throughput - writes (bytes/s)*: change rate of bytes written system-wide
+-  *Host available memory (gigabytes)*: current amount of memory available (not including swap)
+   system-wide
+-  *CPU utilization (percent)*: current utilization of CPU cores, averaged across all cores in the
+   system
+
+For distributed training, these metrics are collected for every agent. The data is broken down by
+agent, and GPU metrics can be further broken down by GPU.
+
+.. note::
+
+   System Metrics record agent-level metrics, so when there are multiple experiments on the same
+   agent, it is difficult to analyze. We suggest that profiling is done with only a single
+   experiment per agent.
+
+.. _how-to-profiling-native-profilers:
+
+***************************
+ Native Training Profilers
+***************************
+
+Sometimes system-level profiling doesn't capture enough data to help debug bottlenecks in model
+training code. Identifying inefficiencies in individual training operations or steps requires a more
+fine-grained context than generic system metrics can provide. For this level of profiling,
+Determined supports integration with training profilers that are native to their frameworks:
+
+-  PyTorch Profiler (:ref:`PyTorch API <pytorch_profiler>`)
+-  DeepSpeed Profiler (:ref:`DeepSpeed API <deepspeed-profiler>`)
+-  TensorFlow Keras Profiler (:ref:`Keras API <keras-profiler>`)
+
+Please see your framework's profiler documentation and the Determined Training API guide for usage
+details.
+
+.. _how-to-profiling-prom-grafana:
+
+************************************
+ Prometheus and Grafana Integration
+************************************
+
+For a more resource-centric view of Determined jobs, Determined provides a Prometheus endpoint along
+with a pre-configured Grafana dashboard. These can be set up to track detailed hardware usage
+metrics for a Determined cluster, and can be helpful for alerting and monitoring cluster health.
+
+The Prometheus endpoint aggregates system metrics and associates them with Determined concepts such
+as experiments, tags, and resource pools, which can be viewed in Grafana. We provide a Grafana
+dashboard that shows real-time resource metrics across an entire cluster as well as experiments,
+containers, and resource pools.
+
```

Comment:
The Prometheus endpoint aggregates system metrics and associates them with Determined concepts such
as experiments, tags, and resource pools, which can be viewed in Grafana. Determined provides a Grafana
dashboard that shows real-time resource metrics across an entire cluster as well as experiments,
containers, and resource pools.